pkg/engine/transform/transform.go

```diff
@@ -353,7 +351,77 @@ func (t *Transformer) NormalizeResourcesForK8sMasterUpgrade(logger *logrus.Entry
 			continue
 		}
 
+		if resourceType == vmssResourceType {
+			filteredDependencies := []string{}
```

Comment:
There is a lot of duplicated code here, could some of this be extracted to a function and reused?